### ISSUE TITLE
6_visualize.py

### ISSUE THREAD

**Issue by SebliLo** [BMIRDS/deepslide#49]
I am running the code in CentOS Linux 7 which is based on RHEL.
How to see the visualization figures like the one you showed in your readme section of "6. Visualization"  Deep Learning Model figures from B.i - B.iv?

py3) [SebliLo@tesla deepslide]$ python code/6_visualize.py
###############     CONFIGURATION     ###############
all_wsi:	all_wsi
val_wsi_per_class:	20
test_wsi_per_class:	30
keep_orig_copy:	True
num_workers:	8
patch_size:	224
wsi_train:	wsi_train
wsi_val:	wsi_val
wsi_test:	wsi_test
labels_train:	labels_train.csv
labels_val:	labels_val.csv
labels_test:	labels_test.csv
train_folder:	train_folder
patches_eval_train:	patches_eval_train
patches_eval_val:	patches_eval_val
patches_eval_test:	patches_eval_test
num_train_per_class:	8000
type_histopath:	True
purple_threshold:	100
purple_scale_size:	15
slide_overlap:	3
gen_val_patches_overlap_factor:	1.5
image_ext:	jpg
by_folder:	True
color_jitter_brightness:	0.5
color_jitter_contrast:	0.5
color_jitter_saturation:	0.5
color_jitter_hue:	0.2
num_epochs:	20
num_layers:	18
learning_rate:	0.001
batch_size:	16
weight_decay:	0.0001
learning_rate_decay:	0.85
resume_checkpoint:	False
save_interval:	1
checkpoints_folder:	checkpoints
checkpoint_file:	xyz.pt
pretrain:	False
log_folder:	logs
auto_select:	True
preds_train:	preds_train
preds_val:	preds_val
preds_test:	preds_test
inference_train:	inference_train
inference_val:	inference_val
inference_test:	inference_test
vis_train:	vis_train
vis_val:	vis_val
vis_test:	vis_test
device:	cpu
classes:	['benign', 'tumor']
num_classes:	2
train_patches:	train_folder/train
val_patches:	train_folder/val
path_mean:	[0.16684997, 0.16684997, 0.16684997]
path_std:	[0.40847272, 0.40847272, 0.40847272]
resume_checkpoint_path:	checkpoints/xyz.pt
log_csv:	logs/log_2212022_231057.csv
eval_model:	checkpoints/xyz.pt
threshold_search:	(0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9)
colors:	('red', 'white', 'blue', 'green', 'purple', 'orange', 'black', 'pink', 'yellow')

#####################################################

+++++ Running 6_visualize.py +++++

----- Visualizing validation set -----
40 whole slides found from wsi_val
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_67_13_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_67_14_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_67_15_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_67_16_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_67_17_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_67_18_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_67_22_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_68_11_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_68_12_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_68_15_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_68_16_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_68_22_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_69_12_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_69_13_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_69_14_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_69_15_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_69_16_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_69_17_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_69_18_0.png of shape (256, 256, 3)
visualizing wsi_val/benign/Copy of 6 3412 2-L-1 HCC_69_19_0.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_104_65_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_104_66_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_105_62_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_105_63_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_105_64_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_105_65_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_105_66_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_105_67_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_105_68_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_49_58_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_50_56_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_51_56_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_52_55_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_59_68_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_59_69_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_60_43_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_60_44_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_60_45_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_60_67_1.png of shape (256, 256, 3)
visualizing wsi_val/tumor/Copy of 6 3412 2-L-1 HCC_61_44_1.png of shape (256, 256, 3)
find the visualizations in vis_val
----- Finished visualizing validation set -----

----- Visualizing test set -----
60 whole slides found from wsi_test
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_70_12_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_70_14_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_70_15_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_70_16_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_70_17_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_70_18_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_70_19_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_71_13_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_71_14_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_71_15_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_71_16_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_71_17_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_71_18_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_71_19_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_72_14_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_72_15_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_72_16_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_72_18_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_72_19_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_72_20_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_72_21_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_73_15_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_73_18_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_73_19_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_73_20_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_74_16_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_74_17_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_74_18_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_75_16_0.png of shape (256, 256, 3)
visualizing wsi_test/benign/Copy of 6 3412 2-L-1 HCC_75_17_0.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_61_45_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_61_46_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_61_66_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_62_42_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_62_43_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_62_46_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_63_41_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_63_42_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_63_44_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_63_45_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_64_40_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_64_43_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_65_42_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_65_51_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_65_52_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_66_50_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_66_51_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_66_52_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_67_50_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_75_8_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_75_9_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_76_6_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_77_7_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_77_8_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_89_64_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_91_64_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_92_62_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_92_63_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_92_64_1.png of shape (256, 256, 3)
visualizing wsi_test/tumor/Copy of 6 3412 2-L-1 HCC_93_65_1.png of shape (256, 256, 3)
find the visualizations in vis_test
----- Finished visualizing test set -----

+++++ Finished running 6_visualize.py +++++

**Comment by SebliLo** [comment]
Plus, to get the model annotated visualization figures, my understanding is the input would be the whole slide, right?

**Comment by SebliLo** [comment]
Or can I manually update the wsi_test and wsi_val folders to have the whole slide images? Currently it looks like wsi_test and  wsi_val have only patches of image

**Comment by JosephDiPalma** [comment]
@SebliLo Currently the code for producing the figures in the paper is unavailable in this repository. To re-create those figures, you would need to write your own function.
